### ISSUE TITLE
replace pep508_rs modern version specifiers

### ISSUE THREAD

**Issue by baszalmstra** [prefix-dev/pixi#466]
The support for "modern" package specifiers in pep508_rs is very underdeveloped. We want to replace that with a custom implementation. For now, we can forgo URL dependencies since those are not supported by pixi. I would like to see an implementation where if a string is specified it must be a version specifier and not a full requirement. If you want to add more complex behavior to your dependency you should create a toml table.

E.g:

```toml
foo = "1.2.3"
foo = { version = "1.2.3", extras=["all"] }
foo = "*"
```

Poetry also supports additional operators that we could translate like [`^` and `~`](https://python-poetry.org/docs/dependency-specification/), lets not do that for now. We can add that later.

Also, errors must be contextual:

```toml
foo = { version="3.2.1", extras=["b$r"] }
```

should result in an error message saying that "b$r" is not a valid extra name. Currently, we produce these types of errors:

```
data did not match any variant of untagged enum RequirementModern
```

**Comment by ruben-arts** [comment]
Is there a specific Pep that includes what you want to have as a version type or what is it precisely combining?

**Comment by baszalmstra** [comment]
Fixed by #471 and #479
